pkg/api/types.go

```diff
@@ -195,6 +195,119 @@ type VolumeSource struct {
 	NFS *NFSVolumeSource `json:"nfs"`
 }
 
+// Similar to VolumeSource but meant for the administrator who creates PVs.
+// Exactly one of its members must be set.
+type PersistentVolumeSource struct {
```

Comment:
Re: the EmptyDir question, with a little code, any volume can likely be persistent.  Otherwise, wouldn't the admin only provision volumes that make sense as persistent volumes?

In other words, a volume is a volume is a volume.  The PV framework around it gives a volume longevity outside a pod, regardless of type of volume.  Splitting them between VS and PVS involves more code and complexity that isn't strictly needed, but I'm not as knowledgeable on the architecture or Go-isms of the project.